cdap-api-spark-base/src/main/java/io/cdap/cdap/api/spark/JavaSparkExecutionContextBase.java

```diff
@@ -268,4 +269,10 @@
    * Returns the underlying {@link SparkExecutionContext} used by this object.
    */
   public abstract SparkExecutionContext getSparkExecutionContext();
+
+  /**
+   *
+   * @return a map of feature flag names and if they are enabled.
+   */
+  public abstract Map<String, Boolean> getFeatureFlags();
```

Comment:
You shouldn't need to add this since it comes from the `FeatureFlagsProvider` interface and you are not implementing it here.